main.go

```diff
@@ -56,11 +56,14 @@ func init() {
 var flags = struct {
 	EnableLeaderElection bool
 	MetricsAddr          string
+	Provider             string
 }{}
 
 func initFlags() {
 	flag.BoolVar(&flags.EnableLeaderElection, "leader-elect", false, "Enable leader election for controller manager.")
 	flag.StringVar(&flags.MetricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&flags.Provider, "provider", "azure", "Provider name for the migration.")
```

Comment:
Why don't we make the provider flag required and avoid a default?
I appreciate the fact you wanted to keep backward compatibility but I think it's best to have the flag fully implemented from the beginning

Reply:
If you are fine with that Than I am more than happy to do it, just did not want to break your  testing

Reply:
go for it, thx